cmd/metricscollector/v1beta1/file-metricscollector/main.go

```diff
@@ -308,7 +308,9 @@ func watchMetricsFile(mFile string, stopRules stopRulesFlag, filters []string) {
 			if err != nil {
 				klog.Fatalf("Could not connect to Early Stopping service, error: %v", err)
 			}
-			defer conn.Close()
+			func () {
+				defer conn.Close()
```

Comment:
It's nice catch!

I think to run it properly, we should add all of the `conn` related code under `func() { ... }()`.
Similar to this comment: https://stackoverflow.com/a/45620423.

Maybe it is better to simple close the connection without `defer` (`conn.Close()`) after the `SetTrialStatus` executes: https://github.com/kubeflow/katib/blob/e9c3afdfb1c4f981b3df2f67bcdfa1a0e5abc57e/cmd/metricscollector/v1beta1/file-metricscollector/main.go#L325

WDYT @tenzen-y ?

Reply:
It makes sense! @andreyvelich 
I fixed it. Could you check the fixed code?

Reply:
Sure, I will take a look at other changes later today.
Thank you updating this!